src/app/myworkflows/my-workflow/my-workflow.component.ts

```diff
@@ -171,12 +171,16 @@ export class MyWorkflowComponent extends MyEntry implements OnInit {
             // If a user navigates directly to an unpublished workflow on their my-workflows page (via bookmark, refresh),
             // the url needs to be used to set the workflow onInit.
             // Otherwise, the select - tab.pipe results in really strange behaviour. Not entirely sure why.
-            this.workflowService.setWorkflow(
-              this.findEntryFromPath(
-                this.urlResolverService.getEntryPathFromUrl(),
-                this.groupEntriesObject.concat(this.groupSharedEntriesObject)
-              )
+            const entry = this.findEntryFromPath(
+              this.urlResolverService.getEntryPathFromUrl(),
+              this.groupEntriesObject.concat(this.groupSharedEntriesObject)
             );
+            if (entry) {
+              // Url could have non-existent entry
+              this.workflowService.setWorkflow(entry);
+              // Call this.selectEntry to fix https://github.com/dockstore/dockstore/issues/2854
+              this.selectEntry(entry);
```

Comment:
this is strange because this.selectEntry most likely calls `this.workflowService.setWorkflow(entry);` resulting in it being called twice

Reply:
Ah yes, what happens is that `this.workflowsService.setWorkflow` from `this.selectEntry`, but as a result of an Ajax request; the test on line 181/185 executes before the request. Will restructure.